scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -69,68 +116,167 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
         };
     }
 
-    List<Issue> runExternalAnalyzers(Project project) {
+    Optional<Map<String, List<Rule>>> getExternalAnalyzers(PrintStream outputStream) {
         StringBuilder newImports = new StringBuilder();
         StringBuilder tomlDependencies = new StringBuilder();
-
-        for (ScanTomlFile.Analyzer analyzer : scanTomlFile.getAnalyzers()) {
-            extractAnalyzerImportsAndDependencies(analyzer, newImports, tomlDependencies);
+        Set<ScanTomlFile.Analyzer> analyzers = scanTomlFile.getAnalyzers();
+        List<String> analyzerDescriptors = new ArrayList<>(analyzers.size());
+        for (ScanTomlFile.Analyzer analyzer : analyzers) {
+            analyzerDescriptors.add(extractAnalyzerImportsAndDependencies(analyzer, newImports, tomlDependencies));
         }
+
         Module defaultModule = project.currentPackage().getDefaultModule();
-        List<String> defaultModuleFiles = defaultModule.documentIds()
-                .stream()
-                .map(documentId -> defaultModule.document(documentId).name())
-                .toList();
-        String documentName;
-        do {
-            documentName = String.format("%s-%s.bal", IMPORT_GENERATOR_FILE, UUID.randomUUID());
-        } while (defaultModuleFiles.contains(documentName));
         ModuleId defaultModuleId = defaultModule.moduleId();
-        DocumentId documentId = DocumentId.create(documentName, defaultModuleId);
-        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), documentName);
+        DocumentId documentId = DocumentId.create(generatedDocumentName, defaultModuleId);
+        DocumentConfig documentConfig = DocumentConfig.from(documentId, newImports.toString(), generatedDocumentName);
         defaultModule.modify().addDocument(documentConfig).apply();
-
         project.currentPackage().ballerinaToml().ifPresent(ballerinaToml -> {
             String tomlFileContent = ballerinaToml.tomlDocument().textDocument().toString();
             ballerinaToml.modify().withContent(tomlFileContent + tomlDependencies).apply();
         });
 
-        List<ScannerContext> scannerContextList = new ArrayList<>();
+        PackageResolution packageResolution = project.currentPackage().getResolution();
+        ResolvedPackageDependency rootPkgNode = new ResolvedPackageDependency(project.currentPackage(),
+                PackageDependencyScope.DEFAULT);
+        List<Package> directDependencies = packageResolution.dependencyGraph().getDirectDependencies(rootPkgNode)
+                .stream().map(ResolvedPackageDependency::packageInstance).toList();
+        Map<String, List<Rule>> externalAnalyzers = new HashMap<>();
+        for (Package pkgDependency : directDependencies) {
+            PackageManifest pkgManifest = pkgDependency.manifest();
+            String org = pkgManifest.org().value();
+            String name = pkgManifest.name().value();
```

Comment:
These variables are not required since it is used in single place.

Reply:
The `org` and `name` variables are used again during the [creation of the in-memory rules](https://github.com/ballerina-platform/static-code-analysis-tool/pull/15/files/8daa975515a8e8826a7c2d48749dce05dac5eb3d#diff-8fbdfbacc9f46bf89493e82b5eb294ea34cf91175dd952080f3bafdc4139d31eR213)